skimage/measure/tests/test_profile.py

```diff
@@ -110,21 +110,21 @@ def test_pythagorean_triangle_transpose_left_down_linewidth():
 def test_reduce_func_mean():
     prof = profile_line(pyth_image, (0, 1), (3, 1), linewidth=3, order=0,
                         reduce_func=np.mean)
-    expected_prof = np.array([0, 0.8, 1, 0.2])
+    expected_prof = pyth_image[:4, :3].mean(-1)
```

Comment:
not sure I understand why this should work...

Reply:
That's exactly what the `reduce_func` argument is supposed to do (see #4206).

Reply:
this I got, but the `pyth_image` is special so that its non-zero pixels correspond exactly to the scan line, do I understand correctly? This last point is not completely clear to me.

Reply:
No, it is just because `src` is set to `(0, 1)` and `dst` to `(3, 1)` with a `linewidth=3`:
`src` and `dst` are indices positioned on the same column (`src[1] = dst[1] = 1`). So the profile line will be computed using 3 (`linewidth`) columns vertically centered around the second column, from the row 0 (`src[0]`) to the row 3 (`dst[0]`).
That's why the expected result is computed using `pyth_image[:4, :3] = pyth_image[src[0]:dst[0]+1, src[1]-1:dst[1]+2]`.
Sorry if this was not clear.

Reply:
oh ok, no sorry, it's just me being slow :-)